File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/CodyAgentServer.kt
@file:Suppress("FunctionName", "ClassName", "RedundantNullable")
package com.sourcegraph.cody.agent.protocol_generated;

import org.eclipse.lsp4j.jsonrpc.services.JsonNotification;
import org.eclipse.lsp4j.jsonrpc.services.JsonRequest;
import java.util.concurrent.CompletableFuture;

@Suppress("unused")
interface CodyAgentServer {
  // ========
  // Requests
  // ========
  @JsonRequest("initialize")
  fun initialize(params: ClientInfo): CompletableFuture<ServerInfo>
  @JsonRequest("shutdown")
  fun shutdown(params: Null?): CompletableFuture<Null?>
  @JsonRequest("chat/new")
  fun chat_new(params: Null?): CompletableFuture<String>
  @JsonRequest("chat/web/new")
  fun chat_web_new(params: Null?): CompletableFuture<Chat_Web_NewResult>
  @JsonRequest("chat/sidebar/new")
  fun chat_sidebar_new(params: Null?): CompletableFuture<Chat_Sidebar_NewResult>
  @JsonRequest("chat/delete")
  fun chat_delete(params: Chat_DeleteParams): CompletableFuture<List<ChatExportResult>>
  @JsonRequest("chat/models")
  fun chat_models(params: Chat_ModelsParams): CompletableFuture<Chat_ModelsResult>
  @JsonRequest("chat/export")
  fun chat_export(params: Chat_ExportParams?): CompletableFuture<List<ChatExportResult>>
  @JsonRequest("chat/import")
  fun chat_import(params: Chat_ImportParams): CompletableFuture<Null?>
  @JsonRequest("chat/setModel")
  fun chat_setModel(params: Chat_SetModelParams): CompletableFuture<Null?>
  @JsonRequest("commands/explain")
  fun commands_explain(params: Null?): CompletableFuture<String>
  @JsonRequest("commands/smell")
  fun commands_smell(params: Null?): CompletableFuture<String>
  @JsonRequest("commands/custom")
  fun commands_custom(params: Commands_CustomParams): CompletableFuture<CustomCommandResult>
  @JsonRequest("customCommands/list")
  fun customCommands_list(params: Null?): CompletableFuture<List<CodyCommand>>
  @JsonRequest("editTask/start")
  fun editTask_start(params: Null?): CompletableFuture<FixupTaskID?>
  @JsonRequest("editTask/accept")
  fun editTask_accept(params: FixupTaskID): CompletableFuture<Null?>
  @JsonRequest("editTask/undo")
  fun editTask_undo(params: FixupTaskID): CompletableFuture<Null?>
  @JsonRequest("editTask/cancel")
  fun editTask_cancel(params: FixupTaskID): CompletableFuture<Null?>
  @JsonRequest("editTask/retry")
  fun editTask_retry(params: FixupTaskID): CompletableFuture<FixupTaskID?>
  @JsonRequest("editTask/getTaskDetails")
  fun editTask_getTaskDetails(params: FixupTaskID): CompletableFuture<EditTask>
  @JsonRequest("editTask/getFoldingRanges")
  fun editTask_getFoldingRanges(params: GetFoldingRangeParams): CompletableFuture<GetFoldingRangeResult>
  @JsonRequest("command/execute")
  fun command_execute(params: ExecuteCommandParams): CompletableFuture<Any>
  @JsonRequest("codeActions/provide")
  fun codeActions_provide(params: CodeActions_ProvideParams): CompletableFuture<CodeActions_ProvideResult>
  @JsonRequest("codeActions/trigger")
  fun codeActions_trigger(params: FixupTaskID): CompletableFuture<FixupTaskID?>
  @JsonRequest("autocomplete/execute")
  fun autocomplete_execute(params: AutocompleteParams): CompletableFuture<AutocompleteResult>
  @JsonRequest("graphql/getRepoIds")
  fun graphql_getRepoIds(params: Graphql_GetRepoIdsParams): CompletableFuture<Graphql_GetRepoIdsResult>
  @JsonRequest("graphql/currentUserId")
  fun graphql_currentUserId(params: Null?): CompletableFuture<String>
  @JsonRequest("featureFlags/getFeatureFlag")
  fun featureFlags_getFeatureFlag(params: FeatureFlags_GetFeatureFlagParams): CompletableFuture<Boolean?>
  @JsonRequest("telemetry/recordEvent")
  fun telemetry_recordEvent(params: TelemetryEvent): CompletableFuture<Null?>
  @JsonRequest("graphql/getRepoIdIfEmbeddingExists")
  fun graphql_getRepoIdIfEmbeddingExists(params: Graphql_GetRepoIdIfEmbeddingExistsParams): CompletableFuture<String?>
  @JsonRequest("graphql/getRepoId")
  fun graphql_getRepoId(params: Graphql_GetRepoIdParams): CompletableFuture<String?>
  @JsonRequest("git/codebaseName")
  fun git_codebaseName(params: Git_CodebaseNameParams): CompletableFuture<String?>
  @JsonRequest("webview/didDispose")
  fun webview_didDispose(params: Webview_DidDisposeParams): CompletableFuture<Null?>
  @JsonRequest("webview/resolveWebviewView")
  fun webview_resolveWebviewView(params: Webview_ResolveWebviewViewParams): CompletableFuture<Null?>
  @JsonRequest("webview/receiveMessageStringEncoded")
  fun webview_receiveMessageStringEncoded(params: Webview_ReceiveMessageStringEncodedParams): CompletableFuture<Null?>
  @JsonRequest("diagnostics/publish")
  fun diagnostics_publish(params: Diagnostics_PublishParams): CompletableFuture<Null?>
  @JsonRequest("testing/progress")
  fun testing_progress(params: Testing_ProgressParams): CompletableFuture<Testing_ProgressResult>
  @JsonRequest("testing/exportedTelemetryEvents")
  fun testing_exportedTelemetryEvents(params: Null?): CompletableFuture<Testing_ExportedTelemetryEventsResult>
  @JsonRequest("testing/networkRequests")
  fun testing_networkRequests(params: Null?): CompletableFuture<Testing_NetworkRequestsResult>
  @JsonRequest("testing/requestErrors")
  fun testing_requestErrors(params: Null?): CompletableFuture<Testing_RequestErrorsResult>
  @JsonRequest("testing/closestPostData")
  fun testing_closestPostData(params: Testing_ClosestPostDataParams): CompletableFuture<Testing_ClosestPostDataResult>
  @JsonRequest("testing/memoryUsage")
  fun testing_memoryUsage(params: Null?): CompletableFuture<Testing_MemoryUsageResult>
  @JsonRequest("testing/heapdump")
  fun testing_heapdump(params: Null?): CompletableFuture<Null?>
  @JsonRequest("testing/awaitPendingPromises")
  fun testing_awaitPendingPromises(params: Null?): CompletableFuture<Null?>
  @JsonRequest("testing/workspaceDocuments")
  fun testing_workspaceDocuments(params: GetDocumentsParams): CompletableFuture<GetDocumentsResult>
  @JsonRequest("testing/diagnostics")
  fun testing_diagnostics(params: Testing_DiagnosticsParams): CompletableFuture<Testing_DiagnosticsResult>
  @JsonRequest("testing/progressCancelation")
  fun testing_progressCancelation(params: Testing_ProgressCancelationParams): CompletableFuture<Testing_ProgressCancelationResult>
  @JsonRequest("testing/reset")
  fun testing_reset(params: Null?): CompletableFuture<Null?>
  @JsonRequest("testing/autocomplete/completionEvent")
  fun testing_autocomplete_completionEvent(params: CompletionItemParams): CompletableFuture<CompletionBookkeepingEvent?>
  @JsonRequest("testing/autocomplete/autoeditEvent")
  fun testing_autocomplete_autoeditEvent(params: CompletionItemParams): CompletableFuture<AutoeditRequestStateForAgentTesting?>
  @JsonRequest("testing/autocomplete/awaitPendingVisibilityTimeout")
  fun testing_autocomplete_awaitPendingVisibilityTimeout(params: Null?): CompletableFuture<CompletionItemID?>
  @JsonRequest("testing/autocomplete/setCompletionVisibilityDelay")
  fun testing_autocomplete_setCompletionVisibilityDelay(params: Testing_Autocomplete_SetCompletionVisibilityDelayParams): CompletableFuture<Null?>
  @JsonRequest("testing/autocomplete/providerConfig")
  fun testing_autocomplete_providerConfig(params: Null?): CompletableFuture<Testing_Autocomplete_ProviderConfigResult?>
  @JsonRequest("extensionConfiguration/change")
  fun extensionConfiguration_change(params: ExtensionConfiguration): CompletableFuture<ProtocolAuthStatus?>
  @JsonRequest("extensionConfiguration/status")
  fun extensionConfiguration_status(params: Null?): CompletableFuture<ProtocolAuthStatus?>
  @JsonRequest("extensionConfiguration/getSettingsSchema")
  fun extensionConfiguration_getSettingsSchema(params: Null?): CompletableFuture<String>
  @JsonRequest("textDocument/change")
  fun textDocument_change(params: ProtocolTextDocument): CompletableFuture<TextDocument_ChangeResult>
  @JsonRequest("attribution/search")
  fun attribution_search(params: Attribution_SearchParams): CompletableFuture<Attribution_SearchResult>
  @JsonRequest("ignore/test")
  fun ignore_test(params: Ignore_TestParams): CompletableFuture<Ignore_TestResult>
  @JsonRequest("testing/ignore/overridePolicy")
  fun testing_ignore_overridePolicy(params: ContextFilters?): CompletableFuture<Null?>
  @JsonRequest("extension/reset")
  fun extension_reset(params: Null?): CompletableFuture<Null?>
  @JsonRequest("internal/getAuthHeaders")
  fun internal_getAuthHeaders(params: String): CompletableFuture<Map<String, String>>

  // =============
  // Notifications
  // =============
  @JsonNotification("initialized")
  fun initialized(params: Null?)
  @JsonNotification("exit")
  fun exit(params: Null?)
  @JsonNotification("extensionConfiguration/didChange")
  fun extensionConfiguration_didChange(params: ExtensionConfiguration)
  @JsonNotification("workspaceFolder/didChange")
  fun workspaceFolder_didChange(params: WorkspaceFolder_DidChangeParams)
  @JsonNotification("textDocument/didOpen")
  fun textDocument_didOpen(params: ProtocolTextDocument)
  @JsonNotification("textDocument/didChange")
  fun textDocument_didChange(params: ProtocolTextDocument)
  @JsonNotification("textDocument/didFocus")
  fun textDocument_didFocus(params: TextDocument_DidFocusParams)
  @JsonNotification("textDocument/didSave")
  fun textDocument_didSave(params: TextDocument_DidSaveParams)
  @JsonNotification("textDocument/didRename")
  fun textDocument_didRename(params: TextDocument_DidRenameParams)
  @JsonNotification("textDocument/didClose")
  fun textDocument_didClose(params: ProtocolTextDocument)
  @JsonNotification("workspace/didDeleteFiles")
  fun workspace_didDeleteFiles(params: DeleteFilesParams)
  @JsonNotification("workspace/didCreateFiles")
  fun workspace_didCreateFiles(params: CreateFilesParams)
  @JsonNotification("workspace/didRenameFiles")
  fun workspace_didRenameFiles(params: RenameFilesParams)
  @JsonNotification("$/cancelRequest")
  fun cancelRequest(params: CancelParams)
  @JsonNotification("autocomplete/clearLastCandidate")
  fun autocomplete_clearLastCandidate(params: Null?)
  @JsonNotification("autocomplete/completionSuggested")
  fun autocomplete_completionSuggested(params: CompletionItemParams)
  @JsonNotification("autocomplete/completionAccepted")
  fun autocomplete_completionAccepted(params: CompletionItemParams)
  @JsonNotification("progress/cancel")
  fun progress_cancel(params: Progress_CancelParams)
  @JsonNotification("testing/runInAgent")
  fun testing_runInAgent(params: String)
  @JsonNotification("webview/didDisposeNative")
  fun webview_didDisposeNative(params: Webview_DidDisposeNativeParams)
  @JsonNotification("secrets/didChange")
  fun secrets_didChange(params: Secrets_DidChangeParams)
  @JsonNotification("window/didChangeFocus")
  fun window_didChangeFocus(params: Window_DidChangeFocusParams)
  @JsonNotification("testing/resetStorage")
  fun testing_resetStorage(params: Null?)
}

File: agent/bindings/csharp/CodyAgentServer.cs
using System.Threading.Tasks;

namespace Cody.Core.Agent.Protocol;
{

public interface CodyAgentServer
{
  // ========
  // Requests
  // ========
  [JsonRpcMethod("initialize")]
  Task<ServerInfo> Initialize(ClientInfo params);
  [JsonRpcMethod("shutdown")]
  Task Shutdown();
  [JsonRpcMethod("chat/new")]
  Task<string> ChatNew();
  [JsonRpcMethod("chat/web/new")]
  Task<ChatWebNewResult> ChatWebNew();
  [JsonRpcMethod("chat/sidebar/new")]
  Task<ChatSidebarNewResult> ChatSidebarNew();
  [JsonRpcMethod("chat/delete")]
  Task<ChatExportResult[]> ChatDelete(ChatDeleteParams params);
  [JsonRpcMethod("chat/restore")]
  Task<string> ChatRestore(ChatRestoreParams params);
  [JsonRpcMethod("chat/models")]
  Task<ChatModelsResult> ChatModels(ChatModelsParams params);
  [JsonRpcMethod("chat/export")]
  Task<ChatExportResult[]> ChatExport(ChatExportParams params);
  [JsonRpcMethod("chat/remoteRepos")]
  Task<ChatRemoteReposResult> ChatRemoteRepos(ChatRemoteReposParams params);
  [JsonRpcMethod("commands/explain")]
  Task<string> CommandsExplain();
  [JsonRpcMethod("commands/test")]
  Task<string> CommandsTest();
  [JsonRpcMethod("commands/smell")]
  Task<string> CommandsSmell();
  [JsonRpcMethod("commands/custom")]
  Task<CustomCommandResult> CommandsCustom(CommandsCustomParams params);
  [JsonRpcMethod("customCommands/list")]
  Task<CodyCommand[]> CustomCommandsList();
  [JsonRpcMethod("editCommands/code")]
  Task<EditTask> EditCommandsCode(EditCommandsCodeParams params);
  [JsonRpcMethod("editCommands/test")]
  Task<EditTask> EditCommandsTest();
  [JsonRpcMethod("editCommands/document")]
  Task<EditTask> EditCommandsDocument();
  [JsonRpcMethod("editTask/accept")]
  Task EditTaskAccept(EditTaskAcceptParams params);
  [JsonRpcMethod("editTask/undo")]
  Task EditTaskUndo(EditTaskUndoParams params);
  [JsonRpcMethod("editTask/cancel")]
  Task EditTaskCancel(EditTaskCancelParams params);
  [JsonRpcMethod("editTask/retry")]
  Task<EditTask> EditTaskRetry(EditTaskRetryParams params);
  [JsonRpcMethod("editTask/getTaskDetails")]
  Task<EditTask> EditTaskGetTaskDetails(EditTaskGetTaskDetailsParams params);
  [JsonRpcMethod("editTask/getFoldingRanges")]
  Task<GetFoldingRangeResult> EditTaskGetFoldingRanges(GetFoldingRangeParams params);
  [JsonRpcMethod("command/execute")]
  Task<Object> CommandExecute(ExecuteCommandParams params);
  [JsonRpcMethod("codeActions/provide")]
  Task<CodeActionsProvideResult> CodeActionsProvide(CodeActionsProvideParams params);
  [JsonRpcMethod("codeActions/trigger")]
  Task<EditTask> CodeActionsTrigger(CodeActionsTriggerParams params);
  [JsonRpcMethod("autocomplete/execute")]
  Task<AutocompleteResult> AutocompleteExecute(AutocompleteParams params);
  [JsonRpcMethod("graphql/getRepoIds")]
  Task<GraphqlGetRepoIdsResult> GraphqlGetRepoIds(GraphqlGetRepoIdsParams params);
  [JsonRpcMethod("graphql/currentUserId")]
  Task<string> GraphqlCurrentUserId();
  [JsonRpcMethod("graphql/currentUserIsPro")]
  Task<bool> GraphqlCurrentUserIsPro();
  [JsonRpcMethod("featureFlags/getFeatureFlag")]
  Task<bool> FeatureFlagsGetFeatureFlag(FeatureFlagsGetFeatureFlagParams params);
  [JsonRpcMethod("graphql/getCurrentUserCodySubscription")]
  Task<CurrentUserCodySubscription> GraphqlGetCurrentUserCodySubscription();
  [JsonRpcMethod("graphql/logEvent")]
  Task GraphqlLogEvent(Event params);
  [JsonRpcMethod("telemetry/recordEvent")]
  Task TelemetryRecordEvent(TelemetryEvent params);
  [JsonRpcMethod("graphql/getRepoIdIfEmbeddingExists")]
  Task<string> GraphqlGetRepoIdIfEmbeddingExists(GraphqlGetRepoIdIfEmbeddingExistsParams params);
  [JsonRpcMethod("graphql/getRepoId")]
  Task<string> GraphqlGetRepoId(GraphqlGetRepoIdParams params);
  [JsonRpcMethod("git/codebaseName")]
  Task<string> GitCodebaseName(GitCodebaseNameParams params);
  [JsonRpcMethod("webview/didDispose")]
  Task WebviewDidDispose(WebviewDidDisposeParams params);
  [JsonRpcMethod("webview/resolveWebviewView")]
  Task WebviewResolveWebviewView(WebviewResolveWebviewViewParams params);
  [JsonRpcMethod("webview/receiveMessageStringEncoded")]
  Task WebviewReceiveMessageStringEncoded(WebviewReceiveMessageStringEncodedParams params);
  [JsonRpcMethod("diagnostics/publish")]
  Task DiagnosticsPublish(DiagnosticsPublishParams params);
  [JsonRpcMethod("testing/progress")]
  Task<TestingProgressResult> TestingProgress(TestingProgressParams params);
  [JsonRpcMethod("testing/exportedTelemetryEvents")]
  Task<TestingExportedTelemetryEventsResult> TestingExportedTelemetryEvents();
  [JsonRpcMethod("testing/networkRequests")]
  Task<TestingNetworkRequestsResult> TestingNetworkRequests();
  [JsonRpcMethod("testing/requestErrors")]
  Task<TestingRequestErrorsResult> TestingRequestErrors();
  [JsonRpcMethod("testing/closestPostData")]
  Task<TestingClosestPostDataResult> TestingClosestPostData(TestingClosestPostDataParams params);
  [JsonRpcMethod("testing/memoryUsage")]
  Task<TestingMemoryUsageResult> TestingMemoryUsage();
  [JsonRpcMethod("testing/awaitPendingPromises")]
  Task TestingAwaitPendingPromises();
  [JsonRpcMethod("testing/workspaceDocuments")]
  Task<GetDocumentsResult> TestingWorkspaceDocuments(GetDocumentsParams params);
  [JsonRpcMethod("testing/diagnostics")]
  Task<TestingDiagnosticsResult> TestingDiagnostics(TestingDiagnosticsParams params);
  [JsonRpcMethod("testing/progressCancelation")]
  Task<TestingProgressCancelationResult> TestingProgressCancelation(TestingProgressCancelationParams params);
  [JsonRpcMethod("testing/reset")]
  Task TestingReset();
  [JsonRpcMethod("testing/autocomplete/completionEvent")]
  Task<CompletionBookkeepingEvent> TestingAutocompleteCompletionEvent(CompletionItemParams params);
  [JsonRpcMethod("extensionConfiguration/change")]
  Task<AuthStatus> ExtensionConfigurationChange(ExtensionConfiguration params);
  [JsonRpcMethod("extensionConfiguration/status")]
  Task<AuthStatus> ExtensionConfigurationStatus();
  [JsonRpcMethod("extensionConfiguration/getSettingsSchema")]
  Task<string> ExtensionConfigurationGetSettingsSchema();
  [JsonRpcMethod("textDocument/change")]
  Task<TextDocumentChangeResult> TextDocumentChange(ProtocolTextDocument params);
  [JsonRpcMethod("attribution/search")]
  Task<AttributionSearchResult> AttributionSearch(AttributionSearchParams params);
  [JsonRpcMethod("ignore/test")]
  Task<IgnoreTestResult> IgnoreTest(IgnoreTestParams params);
  [JsonRpcMethod("testing/ignore/overridePolicy")]
  Task TestingIgnoreOverridePolicy(ContextFilters params);
  [JsonRpcMethod("remoteRepo/has")]
  Task<RemoteRepoHasResult> RemoteRepoHas(RemoteRepoHasParams params);
  [JsonRpcMethod("remoteRepo/list")]
  Task<RemoteRepoListResult> RemoteRepoList(RemoteRepoListParams params);

  // =============
  // Notifications
  // =============
  [JsonRpcMethod("initialized")]
  void Initialized(Void params);
  [JsonRpcMethod("exit")]
  void Exit(Void params);
  [JsonRpcMethod("extensionConfiguration/didChange")]
  void ExtensionConfigurationDidChange(ExtensionConfiguration params);
  [JsonRpcMethod("workspaceFolder/didChange")]
  void WorkspaceFolderDidChange(WorkspaceFolderDidChangeParams params);
  [JsonRpcMethod("textDocument/didOpen")]
  void TextDocumentDidOpen(ProtocolTextDocument params);
  [JsonRpcMethod("textDocument/didChange")]
  void TextDocumentDidChange(ProtocolTextDocument params);
  [JsonRpcMethod("textDocument/didFocus")]
  void TextDocumentDidFocus(TextDocumentDidFocusParams params);
  [JsonRpcMethod("textDocument/didSave")]
  void TextDocumentDidSave(TextDocumentDidSaveParams params);
  [JsonRpcMethod("textDocument/didClose")]
  void TextDocumentDidClose(ProtocolTextDocument params);
  [JsonRpcMethod("workspace/didDeleteFiles")]
  void WorkspaceDidDeleteFiles(DeleteFilesParams params);
  [JsonRpcMethod("workspace/didCreateFiles")]
  void WorkspaceDidCreateFiles(CreateFilesParams params);
  [JsonRpcMethod("workspace/didRenameFiles")]
  void WorkspaceDidRenameFiles(RenameFilesParams params);
  [JsonRpcMethod("$/cancelRequest")]
  void CancelRequest(CancelParams params);
  [JsonRpcMethod("autocomplete/clearLastCandidate")]
  void AutocompleteClearLastCandidate(Void params);
  [JsonRpcMethod("autocomplete/completionSuggested")]
  void AutocompleteCompletionSuggested(CompletionItemParams params);
  [JsonRpcMethod("autocomplete/completionAccepted")]
  void AutocompleteCompletionAccepted(CompletionItemParams params);
  [JsonRpcMethod("progress/cancel")]
  void ProgressCancel(ProgressCancelParams params);
  [JsonRpcMethod("webview/didDisposeNative")]
  void WebviewDidDisposeNative(WebviewDidDisposeNativeParams params);
}
}
